### PR TITLE
add providers to `mixins` folder

### PR DESCRIPTION
**Pull request by Benbentwo**
## what
*  Copies some common providers to the mixins folder

## why
* Have a central place where our common providers are held.

### CODE REVIEW

**Review by max-lobur — APPROVED** [review]

File: mixins/providers-aws-superadmin.tf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+provider "aws" {
+  region = var.region
+}
```
Comment (osterman) [inline mixins/providers-aws-superadmin.tf:3]: Should `var.region` be defined?

File: mixins/provider-datadog.tf [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module "datadog_configuration" {
+  source  = "../datadog-configuration/modules/datadog_keys"
+  region  = var.region
```
Comment (osterman) [inline mixins/provider-datadog.tf:3]: Should `var.region` be defined?

Comment (osterman) [reply mixins/provider-datadog.tf:3]: If not, maybe add a comment, like:
```
# Depends on `var.region`
```

**Comment by Benbentwo** [comment]
regarding `var.region` and it's necessity. I think this warrants a bigger discussion and isn't in scope of this PR which is a reference point for vendoring, of which our current providers use var.region

**Pull request merged**